columnar/src/column_values/mod.rs

```diff
@@ -58,10 +58,21 @@ pub trait ColumnValues<T: PartialOrd = u64>: Send + Sync {
     /// # Panics
     ///
     /// May panic if `idx` is greater than the column length.
-    fn get_vals(&self, idx: &[u32], output: &mut [T]) {
-        assert!(idx.len() == output.len());
-        for (out, idx) in output.iter_mut().zip(idx.iter()) {
-            *out = self.get_val(*idx as u32);
+    fn get_vals(&self, indexes: &[u32], output: &mut [T]) {
+        assert!(indexes.len() == output.len());
+        let mut iter = output.chunks_exact_mut(4).zip(indexes.chunks_exact(4));
+        for (out, idx) in iter {
+            out[0] = self.get_val(idx[0] as u32);
+            out[1] = self.get_val(idx[1] as u32);
+            out[2] = self.get_val(idx[2] as u32);
+            out[3] = self.get_val(idx[3] as u32);
+        }
+
+        let step_size = 4;
+        let cutoff = indexes.len() - indexes.len() % step_size;
+
```

Comment:
ChunksExact has a reminder
[remainder](https://doc.rust-lang.org/std/slice/struct.ChunksExact.html#method.remainder)

Reply:
this case is a little special because of the `zip`